web/packages/teleport/src/Bots/Edit/EditDialog.tsx

```diff
@@ -0,0 +1,298 @@
+  const { isSuccess, data, error, isLoading } = useGetBot(
+    { name: botName },
+    {
+      enabled: hasReadPermission,
+      staleTime: 30_000, // Keep data in the cache for 30 seconds
+    }
+  );
+
+  const {
+    mutate,
+    error: saveError,
+    isPending: isSubmitting,
+  } = useMutation({
+    mutationFn: (params: EditBotRequest) => {
+      return editBot(ctx.getFeatureFlags(), botName, params);
+    },
+    onSuccess: newData => {
+      const key = createGetBotQueryKey({ name: newData.name });
+      queryClient.setQueryData(key, newData);
+
+      setSelectedRoles(null);
+      setSelectedTraits(null);
+      setSelectedMaxSessionDuration(null);
+
+      onSuccess(newData);
+    },
+  });
+
+  const handleSubmit = () => {
+    const roles = selectedRoles ?? null;
+    const traits =
+      selectedTraits?.map(t => ({
+        name: t.traitKey.value,
+        values: t.traitValues.map(v => v.value),
+      })) ?? null;
+    const max_session_ttl =
+      selectedMaxSessionDuration?.trim()?.replaceAll(' ', '') ?? null;
+
+    const request = {
+      roles,
+      traits,
+      max_session_ttl,
+    };
+
+    mutate(request);
+  };
+
+  const isDirty =
+    selectedRoles !== null ||
+    selectedTraits !== null ||
+    selectedMaxSessionDuration !== null;
+
+  return (
+    <Dialog open onClose={onCancel}>
+      <DialogHeader>
+        <DialogTitle>Edit Bot</DialogTitle>
+      </DialogHeader>
+      <Validation>
+        {({ validator }) => (
+          <form
+            onSubmit={e => {
+              e.preventDefault();
+              if (
+                hasEditPermission &&
+                isDirty &&
+                !isSubmitting &&
+                validator.validate()
+              ) {
+                handleSubmit();
+              }
+            }}
+          >
+            <DialogContent maxWidth={680}>
+              {isLoading ? (
+                <Box data-testid="loading" textAlign="center" m={10}>
+                  <Indicator />
+                </Box>
+              ) : undefined}
+
+              {error ? (
+                <Alert kind="danger">{`Error: ${error.message}`}</Alert>
+              ) : undefined}
+
+              {isSuccess && data === null && (
+                <Alert kind="warning">Bot {botName} does not exist</Alert>
+              )}
+
+              {!hasReadPermission ? (
+                <Alert kind="info">
+                  <Flex gap={2}>
+                    You do not have permission to view this bot. Missing role
+                    permissions: <code>bots.read</code>
+                  </Flex>
+                </Alert>
+              ) : undefined}
+
+              {!hasEditPermission ? (
+                <Alert kind="info">
+                  <Flex gap={2}>
+                    You do not have permission to edit this bot. Missing role
+                    permissions: <code>bots.edit</code>
+                  </Flex>
+                </Alert>
+              ) : undefined}
```

Comment:
nit: i think it's kinda weird to render both alerts if they end up not having both perms, could we render one and render a list of missing perms instead?